migration-guides/0.15-Migration-Guide.md

```diff
@@ -446,18 +504,169 @@ case foo of
 
 See [Unbiasing the Semigroup instance for Map](https://discourse.purescript.org/t/unbiasing-the-semigroup-instance-for-map/1935) and [purescript/purescript-ordered-collections#38](https://github.com/purescript/purescript-ordered-collections/pull/38) for more context.
 
-### `purescript-foreign-object`'s `Semigroup` instance was changed
+### Update `frequency` to use `NonEmptyArray` (`purescript-quickcheck`)
+
+In the `0.14.x` ecosystem update, `oneOf` was changed from using a `NonEmptyList` to using a `NonEmptyArray` due to the better syntax support of arrays. However, `frequency` was not also updated. This change fixes that.
+
+**To fix**:
+```diff
+- import Data.List.NonEmpty (NonEmptyList(..))
+- import Data.NonEmpty (NonEmpty(..), (:|))
++ import Data.Array.NonEmpty as NEA
+
+- foo = frequency $ NonEmptyList $ NonEmpty (Tuple 1 (pure 1)) $
+-  (Tuple 2 (pure 2))
+-  : (Tuple 3 (pure 3))
+-  : Nil
+
+{- If you are using a literal non-empty array -}
 
-This section has yet to be written. Including here because it relates to the Map discussion above.
++ import Partial.Unsafe (unsafePartial)
++ import Data.Maybe (fromJust)
+
++ foo = frequency $ unsafePartial $ fromJust $ NEA.fromArray
++  [ Tuple 1 $ pure 1
++  , Tuple 2 $ pure 2
++  , Tuple 3 $ pure 3
++  ]
+
+{- otherwise... -}
+
++ foo = frequency $ NEA.cons' (Tuple 1 (pure 1)) someArrayOfTupleGen
+```
+
+### Drop `Maybe` wrapper and bounds check in `slice` (`purescript-strings`)
+
+`Data.String.CodeUnits.slice` no longer does a bounds check. As such, it no longer returns `Maybe String`, but just `String`.
+```purs
+-- 0.14.x
+slice 0 10 "0123" == Nothing
+slice (-1) 10 "0123" == Nothing
+
+-- 0.15.x
+slice 0 10 "0123" == "0123"
+slice (-1) 10 "0123" == "3"
+```
+
+**To fix**:
+- Account for the lack of `Maybe`
+- Check whether your args are passing in negative indices
 
 ## Breaking Changes in the `purescript-contrib` libraries
 
-This section has yet to be written
+### Make `launchAff_` only work on `Aff Unit` (`purescript-aff`)
+
+`launchAff_ :: forall a. Aff a -> Effect Unit` will _silently_ discard what the `Aff` computation returns. This behavior isn't always desirable, but the developer won't know otherwise.
+
+Thus, `launchAff_`'s type signature was changed to `Aff Unit -> Effect Unit`. By making this change, developers are forced to clarify whether they want this discarding behavior or not. It's a minor annoyance (and breaking change) for most developers but otherwise makes the resulting code much safer.
+
+**To fix**:
+- If the "discarding" behavior is desired, replace `launchAff_ do` with `launchAff_ $ void do`
+- If undesired, now you know where to fix your code
+
+### TODO - Miscellaneous changes in `purescript-css`
```

Comment:
Technically speaking, we only need to include core libraries in the migration guide. We can include libraries outside that if they're particularly widely-used (`aff`, `parsing`), but I don't think we should go much further than that.